homeassistant/components/bthome/sensor.py

```diff
@@ -297,6 +297,16 @@
         native_unit_of_measurement=UnitOfVolumeFlowRate.CUBIC_METERS_PER_HOUR,
         state_class=SensorStateClass.MEASUREMENT,
     ),
+    # Water (L)
+    (
+        BTHomeSensorDeviceClass.WATER,
+        Units.VOLUME_LITERS,
+    ): SensorEntityDescription(
+        key=f"{BTHomeSensorDeviceClass.WATER}_{Units.VOLUME_LITERS}",
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        state_class=SensorStateClass.TOTAL_INCREASING,
```

Comment:
State class `TOTAL_INCREASING` is for sensors which periodically restart their counter from 0. Can you confirm that's the case for this sensor?

Reply:
> it is up to the user how he/she has defined the data

I don't agree, because by setting the state class to `TOTAL_INCREASING` you've locked down the definition of the data such that a dip in the readout will be interpreted as a new meter cycle.

For gas and water this restriction is probably OK because a gas or water utility meter will generally not count backwards.
For an energy meter it is potentially problematic if the meter does not separate import and export but is instead a net meter.

In all cases, `TOTAL_INCREASING` is highly error prone if the source data is not 100% reliable, because a single bad readout of the sensor data will not correct itself, it will instead cause a new meter cycle to be detected.

I think you need to document this very clearly in the bthome [documentation](https://bthome.io/format/), and perhaps consider either of:
- Using state class `TOTAL` instead
- Add some configuration, maybe via new object ids, so bthome sensors can be configured as resetting or non resetting

Reply:
Ok, clear. I will change it to `TOTAL` for now and add a note in the bthome docs.
In the future, I will look into making this configurable.